static/app/components/workflowEngine/ui/collapsibleSection.tsx

```diff
@@ -0,0 +1,75 @@
+import {useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Flex} from 'sentry/components/container/flex';
+import InteractionStateLayer from 'sentry/components/interactionStateLayer';
+import {IconChevron} from 'sentry/icons';
+import {space} from 'sentry/styles/space';
+
+export default function CollapsibleSection({
+  children,
+  title,
+  description,
+  collapsible = true,
+  initialCollapse = false,
+}: {
+  children: React.ReactNode;
+  title: string;
+  collapsible?: boolean;
+  description?: string;
+  initialCollapse?: boolean;
+}) {
+  const [isOpen, setIsOpen] = useState(initialCollapse);
+  const toggleOpen = () => {
+    setIsOpen(!isOpen);
+  };
+
+  return (
+    <Body column>
+      <HeadingWrapper
+        onClick={toggleOpen}
+        justify="space-between"
+        collapsible={collapsible}
+      >
```

Comment:
This renders a `div` so we aren't getting any accessibility roles automatically. The element is not focusable and doesn't surface its state to screen readers.

I'd recommend using the [`details`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) + `summary` elements here, but you could also use a `button` with [`aria-expanded`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-expanded)